test_aodncore/pipeline/steps/test_harvest.py

```diff
@@ -147,15 +147,17 @@ def test_multi_harvest_undo(self):
         multi_harvester_runner = TalendHarvesterRunner(self.uploader, None, TESTDATA_DIR, self.config, self.mock_logger)
         with self.assertRaises(SystemCommandFailedError):
             multi_harvester_runner.run(harvest_collection)
-        self.assertTrue(harvest_collection[0].is_harvest_undone)
+
+        self.assertTrue(all(f.is_harvest_undone for f in harvest_collection))
+        self.assertTrue(all(f.is_storage_undone for f in harvest_collection))
 
     def test_multi_storage_undo(self):
         harvest_collection = get_storage_collection(True)
         os.environ['PIPELINE_TRIGGER_CONFIG_FILE'] = os.path.join(TEST_ROOT, 'trigger_single_fail.conf')
         multi_harvester_runner = TalendHarvesterRunner(self.uploader, None, TESTDATA_DIR, self.config, self.mock_logger)
         with self.assertRaises(SystemCommandFailedError):
             multi_harvester_runner.run(harvest_collection)
-        self.assertTrue(harvest_collection[0].is_storage_undone)
+        self.assertTrue(all(f.is_storage_undone for f in harvest_collection))
```

Comment:
Couldn't the above three test methods be combined into one? They do all the same setup, right through to running the harvest with an expected error. Just need to combine the assertions at the end. No

Reply:
Some are working off a HARVEST_ONLY collection, and some off a HARVEST_UPLOAD collection, so subtly different. I think in general if the setup is cheap, then it's clearer to have separate test methods so it's easier to determine which test actually failed.

Reply:
Ah, I missed that detail. :+1: